src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java

```diff
@@ -181,6 +181,9 @@ public BitbucketSCMRepository getBitbucketSCMRepository() {
     CustomGitSCMSource getAndInitializeGitSCMSourceIfNull() {
         if (gitSCMSource == null) {
             initializeGitScmSource();
+        } 
+        if (getOwner() != null && gitSCMSource.getOwner() == null) {
```

Comment:
Does this only have to happen after the scm source is initialized or are there other scenarios where we might need to set the owner? If the former, should we put this code as part of `initializeGitScmSource`?

Reply:
The GitSCMSource always needs an owner, so it does need to be in the initialization too. It's done as part of constructing the new source:
```
        gitSCMSource = new CustomGitSCMSource(remoteConfig.getUrl(), repository);
        gitSCMSource.setBrowser(new BitbucketServer(selfLink));
        gitSCMSource.setCredentialsId(repository.getCloneCredentialsId());
        gitSCMSource.setOwner(getOwner());
        gitSCMSource.setTraits(traits);
        gitSCMSource.setId(getId() + "-git-scm");
```
One thing to note here is we also set a browser which is not serialized with the rest of the source. I'm actually not 100% sure of what that browser is doing- theoretically it generates links out to source browsers but we already do that manually with `BitbucketLink`. That said I'm happy to include a null check for it as well, for completeness.

Reply:
The browser provides the links for when you view the changeset in Jenkins. Would be good to include it as well. Either in this PR or in a separate PR

Reply:
The browser is a slightly trickier task, as it relies on the self-link which we don't currently persist anywhere. I might tackle this in a separate PR, it's a larger change and unrelated to this issue.

Reply:
I am slightly concerned that we're removing the `browser` but... I think we can address that in a separate PR before release.